src/Android/Services/DeviceActionService.cs

```diff
@@ -378,6 +378,34 @@ public void RateApp()
             }
         }
 
+        public void DisableAutofillService()
+        {
+            try
+            {
+                var activity = (MainActivity)CrossCurrentActivity.Current.Activity;
+                var type = Java.Lang.Class.FromType(typeof(AutofillManager));
+                var manager = activity.GetSystemService(type) as AutofillManager;
```

Comment:
Will the system service of the given `type` always be `AutofillManager` and alway exist? Generally using the `as` keyword for casting is a "safe" cast, so if the type doesn't match or the value returned is `null`, the return result will be `null`; in the line below there's no null check, so assuming it always succeeds, in this scenario it would be more appropriate to use an explicit cast, `var manager = (AutofillManager)activity.GetSystemService(type);` so that it will throw a proper exception if the cast is invalid or the type is missing vs. getting a null ref exception on the next line.

Reply:
It will _not_ always exist since Autofill wasn't introduced until Android 8.  I believe this is the reason the same approach was taken further down in the `SupportsAutofillService()` method, but correct me if I'm wrong @kspearrin .

Reply:
Yes, this is probably throwing on those older devices, but you have a try/catch to avoid the null ref.